lib/opts/index.ts

```diff
@@ -0,0 +1,122 @@
+import { getConfiguration as getConfigFile } from "../configuration";
+import { isin } from "../../type-helpers";
+import type { PrettyPrint } from "../../type-helpers";
+
+type Release = "minor" | "major" | "patch";
+type Task = "changelog" | "commit" | "tag";
+type Hook =
+  | "prerelease"
+  | "prebump"
+  | "postbump"
+  | "prechangelog"
+  | "postchangelog"
+  | "precommit"
+  | "postcommit"
+  | "pretag"
+  | "posttag";
+
+type TypePrefixes = Array<
+  (
+    | { section: string; hidden?: boolean | undefined }
+    | { hidden: true; section?: string | undefined }
+  ) & { type: string }
+>;
+
+type ConfigFiles = Array<
+  | string
+  | { filename: string; type: "json" | "gradle" | "plain-text" }
+  | { filename: string; updater: string }
+>;
+
+/**
+ * __THIS SHOULD NOT CHANGE.__
+ * @deprecated
+ *
+ * The configuration options for `standard-version` as of version 9.5 (The last version prior to the fork; deprecated).
+ */
+export type SVConfig = {
+  packageFiles: ConfigFiles;
+  bumpFiles: ConfigFiles;
+  releaseAs: Release;
+  prerelease: string | boolean;
+  infile: string;
+  message: string;
+  firstRelease: boolean;
+  sign: boolean;
+  noVerify: boolean;
+  commitAll: boolean;
+  silent: boolean;
+  tagPrefix: string;
+  scripts: Record<Hook, string>;
+  skip: Record<Task, string>;
+  dryRun: boolean;
+  gitTagFallback: boolean;
+  path: string;
+  changelogHeader: string;
+  preset: string;
+  lernaPackage: string;
+  header: string;
+  types: TypePrefixes;
+  preMajor: boolean;
+  commitUrlFormat: string;
+  compareUrlFormat: string;
+  issueUrlFormat: string;
+  userUrlFormat: string;
+  releaseCommitMessageFormat: string;
+  issuePrefixes: string[];
+};
+
+/**
+ * The configuration object for commit-and-tag-version, which is a superset of the conventional-changelog-config-spec (as of version 2.1.0)
+ * This may or may not maintain backwards compatibility with standard-version (as of version 9.5.0).
+ */
+export type CatVConfig = PrettyPrint<
+  SVConfig & {
+    npmPublishHint: string;
+    releaseCount: number;
+    tagForce: boolean;
+  }
+>;
+
+/** The configuration options that are not supported by standard-version (as of version 9.5.0). */
+const catVOnlyFeatures = [
+  "npmPublishHint",
+  "releaseCount",
+  "tagForce",
+] as const satisfies ReadonlyArray<Exclude<keyof CatVConfig, keyof SVConfig>>;
+
+export const getMergedConfig = async (
+  cwd?: string
+): Promise<Partial<CatVConfig>> => {
+  const dir = cwd ?? process.cwd();
+  const json = (await import("path")).join(dir, "package.json");
+  const svConfig: SVConfig = (await import(json))["standard-version"];
```

Comment:
These two lines should be typed `SVConfig | undefined` and `CatVConfig | undefined`

Reply:
Also, I'm not sure if the `import` statement will work for importing a user's package.json after transpilation - it might depend on our settings, which means we might have to read the file directly. I'm not sure.

Reply:
Another good catch, will fix in the next commit 😄

Reply:
From my understanding, *dynamic* import statements should work fine in either ESM or CJS, and are evaluated at runtime (not transpile time). Testing will tell for sure, though.

I'll also clean up the uneccessary truthiness check on `svConfig`, since `for...in undefined` is just a no-op.

Reply:
Now that I think about it, I did test this module locally after transpiling. It should work as expected (fingers crossed)

Reply:
I've actually gone ahead and added the union with `undefined` to the type alias definitions: `Config` and `LegacyConfig`. This is more correct - since the configuration supplied by the user can, technically, be undefined. The `Defaults` would fill in any missing options, in that case.